crypto/src/key/mod.rs

```diff
@@ -140,6 +163,81 @@ impl From<MLRistrettoPrivateKey> for PrivateKey {
     }
 }
 
+impl ExtendedPrivateKey {
```

Comment:
Shouldn't there be a method that extracts the private key from the extended private key? How else are we gonna sign stuff?

Reply:
This was a mistake. I have added a `private_key()` method. Unfortunately I couldn't find a way to make it return a reference.

Reply:
> This was a mistake. I have added a `private_key()` method. Unfortunately I couldn't find a way to make it return a reference.

I don't think it's possible to return the final private key as a reference unless you pre-compute the key and store it with in the struct. Otherwise, you get a reference to a local variable which is not allowed. In that case, I'd say just return it by value.